kv/dist_sender.go

```diff
@@ -655,14 +655,17 @@ func (ds *DistSender) Send(_ context.Context, call proto.Call) {
 
 		var desc, descNext *proto.RangeDescriptor
 		err := retry.WithBackoff(retryOpts, func() (retry.Status, error) {
-			var descErr error
+			var err error
 			// Get range descriptor (or, when spanning range, descriptors).
 			// sendAttempt below may clear them on certain errors, so we
 			// refresh (likely from the cache) on every retry.
-			desc, descNext, descErr = ds.getDescriptors(call)
+			desc, descNext, err = ds.getDescriptors(call)
 			// If getDescriptors fails, we don't fish for retryable errors.
-			if descErr != nil {
-				return retry.Break, descErr
+			if err != nil {
+				if rErr, ok := err.(util.Retryable); ok && rErr != nil && rErr.CanRetry() {
```

Comment:
"need" is relative. I'm not sure what the policy on methods with a pointer receiver is, but `(*proto.Error).CanRetry` (and many others) will simply explode if you attempt that, so I'm checking for now. Of course we don't return a `nil` retryable error anywhere. I would like to fix all those pointer receiver issues (and then the check would not be needed) but that's for another PR.